instrgen/rtlib/rtlib.go

```diff
@@ -42,30 +62,109 @@ func NewTracingState() TracingState {
 
 	// Write telemetry data to a file.
 	var err error
-	tracingState.File, err = os.Create("traces.txt")
-
-	if err != nil {
-		tracingState.Logger.Fatal(err)
+	serviceName := os.Getenv(serviceName)
+	// fallback to instrgen
+	if serviceName == "" {
+		serviceName = defaultServiceName
 	}
-	var exp trace.SpanExporter
-	exp, err = NewExporter(tracingState.File)
-	if err != nil {
-		tracingState.Logger.Fatal(err)
+	exporterVar := os.Getenv(tracesExporter)
+	switch exporterVar {
+	case zipkinExporter:
+		exporterEndpoint := os.Getenv(zipkinEndpoint)
+		// fallback to localhost
+		if exporterEndpoint == "" {
+			exporterEndpoint = defaultZipkinEndpoint
+		}
+		exporter, _ := zipkin.New(
+			exporterEndpoint,
+			zipkin.WithLogger(tracingState.Logger),
+		)
+
+		batcher := trace.NewBatchSpanProcessor(exporter)
+
+		tracingState.Tp = trace.NewTracerProvider(
+			trace.WithSpanProcessor(batcher),
+			trace.WithResource(resource.NewWithAttributes(
+				semconv.SchemaURL,
+				semconv.ServiceName(serviceName),
+			)),
+		)
+	case otlpExporter:
+		ctx := context.Background()
+		res, err := resource.New(ctx,
+			resource.WithAttributes(
+				semconv.ServiceNameKey.String(serviceName),
+				semconv.TelemetrySDKLanguageGo,
+			),
+		)
+		if err != nil {
+			tracingState.Logger.Fatal(err)
+		}
+
+		var client otlptrace.Client
+		protocol := os.Getenv(exporterProtocol)
+		exporterEndpoint := os.Getenv(otlpExporterEndpoint)
+
+		if protocol == exporterHTTPProtocol {
+			if exporterEndpoint == "" {
+				exporterEndpoint = defaultHTTPEndpoint
+			}
+
+			client = otlptracehttp.NewClient(
+				otlptracehttp.WithInsecure(),
+				otlptracehttp.WithEndpoint(exporterEndpoint),
+			)
+		} else {
+			if exporterEndpoint == "" {
+				exporterEndpoint = defaultGrpcEndpoint
+			}
+
+			client = otlptracegrpc.NewClient(
+				otlptracegrpc.WithInsecure(),
+				otlptracegrpc.WithEndpoint(exporterEndpoint),
+			)
+		}
+		traceExporter, err := otlptrace.New(
+			context.Background(),
+			client,
+		)
+
+		if err != nil {
+			tracingState.Logger.Fatal(err)
+		}
+
+		bsp := trace.NewBatchSpanProcessor(traceExporter)
+		tracingState.Tp = trace.NewTracerProvider(
+			trace.WithSampler(trace.AlwaysSample()),
+			trace.WithResource(res),
+			trace.WithSpanProcessor(bsp),
+		)
+	default:
+		// fallback to file exporting
+		tracingState.File, err = os.Create(traceFile)
```

Comment:
Having traces end up in a local file that is not in any standardize format doesn't seem ideal to be the default. Why not have the OTLP exporter be the default?

Reply:
Yes, you are right. I just left local file as it was available from the beginning.